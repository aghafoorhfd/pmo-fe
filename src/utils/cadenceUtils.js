import moment from 'moment';
import { DATE_FORMAT_MMM_D_YEAR, DATE_FORMAT_MMM_D } from 'constants/DateConstant';

export const preProcessingQuartersData = (quarters = []) => {
  const quartersRange = { id: 'quarters' };
  const quartersNames = [];
  quarters.forEach(({ startDate, endDate, name }, index) => {
    quartersNames.push(name);
    quartersRange[`quarter${index}`] = `${moment(startDate).format(DATE_FORMAT_MMM_D_YEAR)} - ${moment(endDate).format(DATE_FORMAT_MMM_D_YEAR)}`;
  });
  return [quartersRange, quartersNames];
};

export const preProcessingSprintsData = (sprints = []) => {
  const obj = { id: 'sprints' };
  if (sprints.length > 0) {
    const { name } = sprints[0] || {};
    let i = name?.split(' ')[1];
    sprints.forEach(({ startDate, endDate }) => {
      obj[`sprint${i}`] = `${moment(startDate).format(DATE_FORMAT_MMM_D)} - ${moment(endDate).format(DATE_FORMAT_MMM_D)}`;
      i++;
    });
  }
  return [obj];
};
