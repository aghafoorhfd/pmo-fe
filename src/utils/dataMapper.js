import { startCase } from 'lodash';

export const chartDataMapper = (data) => {
  const series = [];
  const label = [];
  if (Object.keys(data)?.length) {
    Object.keys(data).forEach((key) => {
      series.push(data[key]);
      label.push(startCase(key));
    });
  }
  return { series, label };
};

export const resourceCommitmentRawData = {
  ranges: {
    244: {
      startDate: '13 Feb',
      endDate: '3 Mar'
    },
    245: {
      startDate: '4 Mar',
      endDate: '12 Mar'
    },
    246: {
      startDate: '13 Feb',
      endDate: '3 Mar'
    },
    247: {
      startDate: '4 Mar',
      endDate: '12 Mar'
    },
    248: {
      startDate: '13 Feb',
      endDate: '3 Mar'
    },
    249: {
      startDate: '4 Mar',
      endDate: '12 Mar'
    },
    250: {
      startDate: '13 Feb',
      endDate: '3 Mar'
    },
    251: {
      startDate: '4 Mar',
      endDate: '12 Mar'
    },
    252: {
      startDate: '13 Feb',
      endDate: '3 Mar'
    },
    253: {
      startDate: '4 Mar',
      endDate: '12 Mar'
    },
    254: {
      startDate: '13 Feb',
      endDate: '3 Mar'
    },
    255: {
      startDate: '4 Mar',
      endDate: '12 Mar'
    }
  },
  teamList: [{ label: 'Ninja Warrior', value: 1 }, { label: 'Fahad Ali', value: 2 }, { label: 'Muhazzib Ali', value: 3 }],
  teams: [{
    teamName: 'Ninja Warrior',
    teamAvailableHours: {
      244: 60,
      245: 80
    },
    availableCapacity: {
      244: '0%',
      245: '40%'
    },
    teamResources: [
      {
        name: 'Joseph Smith',
        availableCapacity: {
          244: '0%',
          245: '80%'
        },
        availableHours: {
          244: '0',
          245: '40'
        },
        projects: {
          'New Website Page Project': {
            244: '20',
            245: '40'
          },
          'Data Migration Project': {
            244: '20',
            245: null
          },
          'Financial Planner': {
            244: '20',
            245: '40'
          },
          'Triangular Market': {
            244: '20',
            245: '40'
          }
        }
      },
      {
        name: 'John',
        availableCapacity: {
          244: '0%',
          245: '80%'
        },
        availableHours: {
          244: '0',
          245: '40'
        },
        projects: {
          'New Website Page Project': {
            244: '20',
            245: '40'
          },
          'Data Migration Project': {
            244: '20',
            245: null
          },
          'Financial Planner': {
            244: '20',
            245: '40'
          },
          'Triangular Market': {
            244: '20',
            245: '40'
          }
        }
      }
    ]

  }
  ]
};
