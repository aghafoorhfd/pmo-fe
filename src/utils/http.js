export const transformError = (error) => {
  let err = error;
  if (error?.response?.data?.errors) {
    err = error.response.data.errors;
    if (Array.isArray(err)) {
      [err] = err;
    }
  }
  return err;
};
