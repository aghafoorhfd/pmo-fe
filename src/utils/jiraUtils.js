const JIRA_STATUS = {
  BACKLOG: 'Backlog',
  TODO: 'To Do',
  INPROGRESS: 'In Progress',
  DONE: 'Done'

};

export const getJiraIssueMap = (jiraIssues = {}) => Object.keys(jiraIssues).map((key) => (
  { issueKey: key, ...jiraIssues[key] })) || [];

export const getJiraStories = (jiraIssues = []) => (
  jiraIssues?.filter((issue) => !(issue?.status === JIRA_STATUS.BACKLOG
        || (issue?.status === JIRA_STATUS.TODO && issue?.assignedToSprints?.length <= 0))));

export const getJiraBacklog = (jiraIssues = []) => (
  jiraIssues?.filter((issue) => (
    [JIRA_STATUS.BACKLOG, JIRA_STATUS.INPROGRESS, JIRA_STATUS.DONE].includes(issue?.status)
            || (issue?.status === JIRA_STATUS.TODO && issue?.assignedToSprints?.length <= 0)
  ))
);
