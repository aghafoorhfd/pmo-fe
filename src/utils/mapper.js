import { get, cloneDeep } from 'lodash';

/**
 * generic mapper that can transform any object
 * @param {object} mapping Required -> Schema for required object with mapping of data object
 * @param {object} data Required -> Used for map data attributes
 * @param {object} defaults Optional -> used to get default value if value not found in data object
 * @returns transformedObject
 */

export const mapper = (mapping, data, defaults = {}) => {
  const transformedObject = cloneDeep(mapping);

  const resolver = (path) => {
    let val = null;
    if (typeof path === 'function') {
      val = path(data);
    } else {
      val = get(data, path);
      if (val === undefined) {
        val = Object.prototype.hasOwnProperty.call(defaults, path) ? defaults[path] : path;
      }
    }
    return val;
  };

  (function iterator(_map) {
    const map = _map;
    Object.keys(map).forEach((key) => {
      if (typeof map[key] === 'object') {
        iterator(map[key]);
      } else {
        map[key] = resolver(map[key]);
      }
    });
  }(transformedObject));

  return transformedObject;
};
