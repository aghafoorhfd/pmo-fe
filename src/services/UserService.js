import fetch from 'auth/FetchInterceptor';
import { AUTH_TOKEN } from 'constants/AuthConstant';
import { USER } from 'constants/RolesConstant';
import { removeItemFromLocalStorage } from 'utils/utils';
import {
  AUTH_SERVICE, FORGOT_PASSWORD, REGISTER_USER, VERIFY_EMAIL, SET_CREDENTIALS,
  USER_SERVICE, LOGIN, RESET_CREDENTIALS, GLOBAL_SERVICE, RESOURCE_TEAM, USER_PROFILE
} from 'constants/ApiConstant';

import { ROLES_ACCESS_TYPES, SELECTED_PACKAGE_KEY } from 'constants/MiscConstant';

const {
  CONFLICT_MANAGER, GENERAL_USER, PROJECT_MANAGER, RESOURCE_MANAGER
} = ROLES_ACCESS_TYPES;

const UserService = {
  login(data) {
    const { email, password, tenantId } = data;
    return fetch({
      url: LOGIN,
      method: 'post',
      data: { userName: email.toLowerCase(), password },
      headers: { 'X-TENANT-ID': tenantId }
    });
  },

  register(data, params) {
    const {
      tenantId, firstName, lastName, email, organizationName
    } = data;
    return fetch({
      url: REGISTER_USER,
      method: 'POST',
      data: {
        firstName, lastName, email: email.toLowerCase(), organizationName
      },
      headers: { 'X-TENANT-ID': tenantId },
      params
    });
  },

  logout() {
    const keysToRemove = [AUTH_TOKEN, USER, SELECTED_PACKAGE_KEY];
    keysToRemove.forEach((k) => removeItemFromLocalStorage(k));
  },

  forgotPassword(data) {
    const { tenantId, email } = data;
    return fetch({
      url: FORGOT_PASSWORD,
      method: 'POST',
      data: email.toLowerCase(),
      headers: { 'X-TENANT-ID': tenantId }
    });
  },

  verifySetPasswordToken(tenantId, token) {
    return fetch({
      url: `${VERIFY_EMAIL}?token=${token}`,
      method: 'get',
      headers: { 'X-TENANT-ID': tenantId }
    });
  },

  setPassword(data) {
    const { tenantId } = data;
    return fetch({
      url: data.newUser ? SET_CREDENTIALS : RESET_CREDENTIALS,
      method: data.newUser ? 'post' : 'put',
      data,
      headers: { 'X-TENANT-ID': tenantId }
    });
  },

  updatePassword(data) {
    return fetch({
      url: `${AUTH_SERVICE}/credentials`,
      method: 'put',
      data
    });
  },

  getPrivileges() {
    return fetch({
      url: `${USER_SERVICE}/privileges`,
      method: 'get'
    });
  },

  getAllUsers(accessType, registrationStatus, page, size) {
    return fetch({
      url: `${USER_SERVICE}/${page}/${size}?accessType=${accessType}&filterAnd=status%7C${registrationStatus}`,
      method: 'get'
    });
  },

  addUser(data) {
    return fetch({
      url: USER_SERVICE,
      method: 'post',
      data
    });
  },

  editUser(data) {
    return fetch({
      url: `${USER_SERVICE}/${data?.id}`,
      method: 'put',
      data
    });
  },

  getGlobalConfigurations() {
    return fetch({
      url: `${GLOBAL_SERVICE}/configurations/project`,
      method: 'get'
    });
  },

  sendInvite(userId) {
    return fetch({
      url: `${USER_SERVICE}/${userId}/invite`,
      method: 'put'
    });
  },

  getUsersStatistics(statType) {
    return fetch({
      url: `${USER_SERVICE}/category/stats?stats=${statType}`,
      method: 'get'
    });
  },

  getUsersByType(accessType) {
    return fetch({
      url: `${USER_SERVICE}?accessType=${accessType}`,
      method: 'get'
    });
  },

  getUsersList() {
    const query = new URLSearchParams();
    query.append('accessType', [
      PROJECT_MANAGER.key,
      CONFLICT_MANAGER.key,
      GENERAL_USER.key,
      RESOURCE_MANAGER.key]);
    return fetch({
      url: `${USER_SERVICE}?${query.toString()}`,
      method: 'get'
    });
  },

  getTeamNames() {
    return fetch({
      url: `${RESOURCE_TEAM}/names`,
      method: 'get'
    });
  },

  getUserProfileData() {
    return fetch({
      url: USER_PROFILE,
      method: 'get'
    });
  },

  revokeUser(userId) {
    return fetch({
      url: `${USER_SERVICE}/${userId}/revoke`,
      method: 'put'
    });
  },

  reActiveUser(userId) {
    return fetch({
      url: `${USER_SERVICE}/${userId}/reactivate`,
      method: 'put'
    });
  },

  rejectUser(userId) {
    return fetch({
      url: `${USER_SERVICE}/${userId}/reject`,
      method: 'put'
    });
  },

  getCompanyData() {
    return fetch({
      url: `${USER_SERVICE}/company`,
      method: 'get'
    });
  }

};
export default UserService;
