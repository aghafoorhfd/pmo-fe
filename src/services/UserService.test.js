import fetch from 'auth/FetchInterceptor';
import UserService from 'services/UserService';
import { removeItemFromLocalStorage } from 'utils/utils';
import {
  AUTH_SERVICE, VERIFY_EMAIL, FORGOT_PASSWORD, REGISTER_USER, SET_CREDENTIALS,
  LOGIN, RESET_CREDENTIALS, GLOBAL_SERVICE, USER_SERVICE, USER_PROFILE
} from 'constants/ApiConstant';

jest.mock('auth/FetchInterceptor');
jest.mock('utils/utils');

describe('Auth Service', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should call login handler ', () => {
    UserService.login({ email: 'test@gmail.com', password: '12345678' });

    expect(fetch).toBeCalledWith({
      data: { email: 'test@gmail.com', password: '12345678' },
      method: 'post',
      url: LOGIN
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call register handler ', () => {
    UserService.register({ firstName: 'John', lastName: 'Dunk', email: 'test@gmail.com' });

    expect(fetch).toBeCalledWith({
      data: { firstName: 'John', lastName: 'Dunk', email: 'test@gmail.com' },
      method: 'POST',
      url: REGISTER_USER
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call sign out handler ', () => {
    UserService.logout();

    expect(removeItemFromLocalStorage).toHaveBeenCalledTimes(2);
    expect(removeItemFromLocalStorage).toMatchSnapshot();
  });

  it('should call forgot password handler ', () => {
    UserService.forgotPassword({ email: 'test@gmail.com' });

    expect(fetch).toBeCalledWith({
      data: { email: 'test@gmail.com' },
      method: 'POST',
      url: `${FORGOT_PASSWORD}`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call verify set password token handler ', () => {
    UserService.verifySetPasswordToken('qwertyuiopasdfghj123456sdgh');

    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${VERIFY_EMAIL}?token=qwertyuiopasdfghj123456sdgh`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call set password handler ', () => {
    const data = {
      password: '123456',
      userId: 'qwertyuiopasdfghj123456sdgh',
      newUser: true
    };
    UserService.setPassword(data);

    expect(fetch).toBeCalledWith({
      data,
      method: 'post',
      url: SET_CREDENTIALS
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call set password handler with reset endpoint ', () => {
    const data = {
      password: '123456',
      userId: 'qwertyuiopasdfghj123456sdgh',
      newUser: false
    };
    UserService.setPassword(data);

    expect(fetch).toBeCalledWith({
      data,
      method: 'put',
      url: RESET_CREDENTIALS
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call getPrivileges handler ', () => {
    UserService.getPrivileges('admin');

    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call getCurrentUsers handler ', () => {
    UserService.getAllUsers({ accessType: 's', registrationStatus: 's' });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call getGlobalConfigurations endpoint ', () => {
    UserService.getGlobalConfigurations();
    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${GLOBAL_SERVICE}/configurations/project`
    });
  });
  it('should call getUsersStatistics handler ', () => {
    UserService.getUsersStatistics();

    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call send invite endpoint ', () => {
    const userId = '1270d309-c262-4e3a-9c76-db6cc';
    UserService.sendInvite(userId);

    expect(fetch).toBeCalledWith({
      method: 'put',
      url: `${USER_SERVICE}/${userId}/invite`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call for user profile data', () => {
    UserService.getUserProfileData();
    expect(fetch).toBeCalledWith({
      method: 'get',
      url: USER_PROFILE
    });
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call update password service', () => {
    UserService.updatePassword();
    expect(fetch).toBeCalledWith({
      method: 'put',
      url: `${AUTH_SERVICE}/credentials`
    });
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call get company data ', () => {
    UserService.getCompanyData();

    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${USER_SERVICE}/company`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call reject user api', () => {
    const userId = '1270d309-c262-4e3a-9c76-db6cc';

    UserService.rejectUser(userId);
    expect(fetch).toBeCalledWith({
      method: 'put',
      url: `${USER_SERVICE}/${userId}/reject`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
});
