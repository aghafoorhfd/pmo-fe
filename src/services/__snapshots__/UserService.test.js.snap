// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Service should call for user profile data 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "http://10.4.100.183:8085/user_service/api/users/user-profile",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call forgot password handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "email": "test@gmail.com",
        },
        "method": "POST",
        "url": "http://10.4.100.183:8085/user_service/api/users/forgot-password",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call get company data  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "http://10.4.100.183:8085/user_service/api/users/company",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call getCurrentUsers handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "http://10.4.100.183:8085/user_service/api/users/undefined/undefined?accessType=[object Object]&filterAnd=status%7Cundefined",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call getPrivileges handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "http://10.4.100.183:8085/user_service/api/users/admin/privileges",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call getProjectConfigurations endpoint  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "undefined/project_service/api/projects/metrics",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call getUsersStatistics handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "http://10.4.100.183:8085/user_service/api/users/category/stats",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call login handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "email": "test@gmail.com",
          "password": "12345678",
        },
        "method": "post",
        "url": "http://10.4.100.183:8080/auth_service/api/users/login",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call register handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "email": "test@gmail.com",
          "firstName": "John",
          "lastName": "Dunk",
        },
        "method": "POST",
        "url": "http://10.4.100.183:8085/user_service/api/users/register",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call reject user api 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "put",
        "url": "http://10.4.100.183:8085/user_service/api/users/1270d309-c262-4e3a-9c76-db6cc/reject",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call send invite endpoint  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "put",
        "url": "http://10.4.100.183:8085/user_service/api/users/1270d309-c262-4e3a-9c76-db6cc/invite",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call set password handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "newUser": true,
          "password": "123456",
          "userId": "qwertyuiopasdfghj123456sdgh",
        },
        "method": "post",
        "url": "http://10.4.100.183:8085/user_service/api/users/credentials",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call set password handler with reset endpoint  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "newUser": false,
          "password": "123456",
          "userId": "qwertyuiopasdfghj123456sdgh",
        },
        "method": "put",
        "url": "http://10.4.100.183:8080/auth_service/api/users/reset-password",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call sign out handler  1`] = `
[MockFunction] {
  "calls": [
    [
      "auth_token",
    ],
    [
      "user",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call update password service 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": undefined,
        "method": "put",
        "url": "http://10.4.100.183:8080/auth_service/api/users/credentials",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call verify set password token handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "http://10.4.100.183:8085/user_service/api/users/verify-email?token=qwertyuiopasdfghj123456sdgh",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service updateProjectMetrics should call updateProjectMetrics endpoint with Post http method 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "metric": [
            "stages",
          ],
        },
        "method": "post",
        "url": "undefined/project_service/api/projects/metrics?metric=PROJECT_STAGES",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service updateProjectMetrics should call updateProjectMetrics endpoint with put http method 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "metric": [
            "stages",
          ],
        },
        "method": "put",
        "url": "undefined/project_service/api/projects/metrics/PROJECT_STAGES",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
